sdk/python/kubeflow/trainer/hf_llm_training.py

```diff
@@ -110,6 +110,13 @@ def load_and_preprocess_data(dataset_dir, transformer_type, tokenizer):
 def setup_peft_model(model, lora_config):
     # Set up the PEFT model
     lora_config = LoraConfig(**json.loads(lora_config))
+    reference_lora_config = LoraConfig()
```

Comment:
Can you help me understand how would `reference_lora_config` will be populated with correct types with Katib tune API? I believe Katib arguments are coming with `--lora_config` which only populates `lora_config`

Reply:
@nsingl00 The logic is quite similar to the process of `TrainingArguments`. When using the Katib tune API, `lora_config` will be populated with parameters passed through "--lora_config". However, these parameters might not have the correct types (e.g., `r` might be a string instead of an integer). This code ensures that after Katib populates `lora_config`, its attributes are converted to the correct types by comparing them with the default values in `reference_lora_config`. This ensures that `lora_config` is correctly configured with the appropriate types for further use. Does this clarify your question?

Reply:
makes sense